staging/src/k8s.io/kubectl/pkg/cmd/explain/explain.go

```diff
@@ -54,28 +54,74 @@ var (
 		kubectl explain pods.spec.containers`))
 )
 
-type ExplainOptions struct {
+// ExplainFlags directly reflect the information that CLI is gathering via flags.
+// They will be converted to Options, which reflect the runtime requirements
+// for the command.
+type ExplainFlags struct {
+	Factory    cmdutil.Factory
```

Comment:
Same comment as in https://github.com/kubernetes/kubernetes/pull/112714, do not embed `Factory` in `ExplainFlags` that's wrong. It can be passed when invoking `ToOptions`, because it looks like we can't easily get rid of it here, yet.